app/policies/user_policy.rb

```diff
@@ -22,27 +22,65 @@ def index?
   def edit_by_project_admin?
     project_admin? && record.portal_teacher
   end
+
   def update_by_project_admin?
     project_admin? && record.portal_teacher
   end
 
+  def project_admin_for_user?
+    return false unless record.respond_to? :cohorts
+    (user.admin_for_project_cohorts & record.cohorts).length > 0
+  end
+
   def show?
     changeable?
   end
 
+  def edit_by_admin?
+    admin_or_manager?
+  end
+
+  def teacher_page?
+    project_admin_for_user? || admin_or_manager?
```

Comment:
I would love it if this worked. However the last time I checked the 'teacher_page' view didn't correctly handle things when a user was just a project admin for one of the teachers projects. So I'm pretty sure that previously this was limited to just admin_or_manager. It also looks like this policy method is not called anywhere, but perhaps I'm missing it.